core/src/main/scala/sttp/tapir/EndpointIO.scala

```diff
@@ -60,6 +60,7 @@ object EndpointTransput {
       schema(_.modifyUnsafe[U](Schema.ModifyCollectionElements)(_.validate(v)))
 
     def description(d: String): ThisType[T] = copyWith(codec, info.description(d))
+    def required(): ThisType[T] = copyWith(codec, info.required(true))
```

Comment:
the optionality is available via `schema.isOptional`, so I don't think we need an additional field here

Reply:
From what I checked it does not work in case of primitive types on for fields nested in objects